DMCompiler/Program.cs

```diff
@@ -14,13 +16,29 @@
 namespace DMCompiler {
     class Program {
         public static int _errorCount = 0;
+        public static string[] CompilerArgs;
+        public static List<string> CompiledFiles = new List<string>(1);
 
         static void Main(string[] args) {
             if (!VerifyArguments(args)) return;
+            CompilerArgs = args;
 
             DateTime startTime = DateTime.Now;
 
-            DMPreprocessor preprocessor = Preprocess(args);
+            DMPreprocessor preprocessor = Preprocess(CompiledFiles);
+            if (HasArgument("--dump-preprocessor"))
+            {
+                StringBuilder result = new();
+                foreach (Token t in preprocessor.GetResult()) {
+                    result.Append(t.Text);
+                }
+
+                string output = Path.ChangeExtension(CompiledFiles[0], "txt") ?? Path.Join(System.AppDomain.CurrentDomain.BaseDirectory, "preprocessor_dump.txt");
```

Comment:
Is there a situation where `Path.ChangeExtension(CompiledFiles[0], "txt")` is null? Also, the extension should instead be changed to `.dm`.

Reply:
```suggestion
                string output = Path.ChangeExtension(CompiledFiles[0], "txt") ?? Path.Join(System.AppDomain.CurrentDomain.BaseDirectory, "preprocessor_dump.dm");
```

Reply:
Practically speaking, no. But the IDE will complain about a possible null without this.